tests/rptest/clients/rpk.py

```diff
@@ -329,29 +333,45 @@ def parse_partition(string):
                                      client_id=m['client_id'],
                                      host=m['host'])
 
-        cmd = ["describe", group]
-        out = self._run_group(cmd)
-        lines = out.splitlines()
-
-        group_name = parse_field("GROUP", lines[0])
-        coordinator = parse_field("COORDINATOR", lines[1])
-        state = parse_field("STATE", lines[2])
-        balancer = parse_field("BALANCER", lines[3])
-        members = parse_field("MEMBERS", lines[4])
-        partitions = []
-        for l in lines[5:]:
-            #skip header line
-            if l.startswith("TOPIC") or len(l) < 2:
-                continue
-            p = parse_partition(l)
-            partitions.append(p)
-
-        return RpkGroup(name=group_name,
-                        coordinator=int(coordinator),
-                        state=state,
-                        balancer=balancer,
-                        members=int(members),
-                        partitions=partitions)
+        def try_describe_group(group):
+            cmd = ["describe", group]
+            out = self._run_group(cmd)
+            lines = out.splitlines()
+
+            group_name = parse_field("GROUP", lines[0])
+            coordinator = parse_field("COORDINATOR", lines[1])
+            state = parse_field("STATE", lines[2])
+            balancer = parse_field("BALANCER", lines[3])
+            members = parse_field("MEMBERS", lines[4])
+            partitions = []
+            for l in lines[5:]:
+                #skip header line
+                if l.startswith("TOPIC") or len(l) < 2:
+                    continue
+                p = parse_partition(l)
+                # p is None if the leader of the partition has changed.
+                if p is None:
+                    return None
+
+                partitions.append(p)
+
+            return RpkGroup(name=group_name,
+                            coordinator=int(coordinator),
+                            state=state,
+                            balancer=balancer,
+                            members=int(members),
+                            partitions=partitions)
+
+        attempts = 10
+        rpk_group = None
+        # try to wait for leadership to stabilize.
+        while rpk_group is None and attempts != 0:
+            rpk_group = try_describe_group(group)
```

Comment:
Retry loops should have a small sleep in (e.g. 0.5s would be good here) to account for the time taken when system is in an election state (elections take somewhere from the 100ms to few-seconds range) -- otherwise there is a risk we blow through our retry could too quickly.